### PR TITLE
Add tests for DNS Module

### PR DESCRIPTION
**Pull request by jackyzha0**
Part of https://github.com/hyphacoop/distributed-press-organizing/issues/65

### CODE REVIEW

File: v1/package.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
     "dev": "ts-node-esm index.ts | pino-pretty -c -t",
     "watch": "nodemon --watch './**/*.ts' --exec 'node --experimental-specifier-resolution=node --loader ts-node/esm' index.ts | pino-pretty -c -t",
     "start": "node build/index.js",
-    "test": "ava --timeout=5m"
+    "test": "ava --timeout=2m"
```
Comment (RangerMauve) [inline v1/package.json:16]:
```suggestion
    "test": "ava --timeout=5m"
```

Sadly IPNS updates take forever so I think we should keep the timeout at 5 minutes to be safe. Maybe 4 minutes (3 mins means we'll get false negatives sometimes)

File: v1/protocols/hyper.ts [hunk 2]
```diff
@@ -64,11 +64,13 @@ export class HyperProtocol implements Protocol<Static<typeof HyperProtocolFields
 
     ctx?.logger.info(`[hyper] Published: ${link}`)
 
+    const dnslink = raw.substring('hyper://'.length)
```
Comment (RangerMauve) [inline v1/protocols/hyper.ts:67]:
```suggestion
    const dnslink = `/hyper/${raw.substring('hyper://'.length)}/`
```

Unless I'm mistaken, we need the `/hyper/` prefix added here.

File: v1/protocols/ipfs.ts [hunk 3]
```diff
@@ -166,7 +166,8 @@ export class IPFSProtocol implements Protocol<Static<typeof IPFSProtocolFields>>
       // TODO: Pass in gateway parameters in options (DP domain name?)
       gateway: `https://gateway.ipfs.io/ipns/${publishKey}/`,
       cid,
-      pubKey
+      pubKey,
+      dnslink: `/ipns/${publishKey}`
```
Comment (RangerMauve) [inline v1/protocols/ipfs.ts:170]:
```suggestion
      dnslink: `/ipns/${publishKey}/`
```

IIRC there needs to be a trailing `/` in paths.

File: v1/protocols/hyper.ts (hunk 2 above)
Comment (jackyzha0) [reply v1/protocols/hyper.ts:67]: ah good catch! not sure how I didn't get that earlier

**Pull request merged**